File: Java/1009. Complement of Base 10 Integer.java
class Solution {
    public int bitwiseComplement(int n) {
        if(n == 0){
            return 1;
        }
        
        int record = n, bit = 1;
        while(record > 0){
            n ^= bit;
            bit <<= 1;
            record >>= 1;
        }
        
        return n;
    }
}

File: Python3/1009. Complement of Base 10 Integer.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        
        if n == 0:
            return 1
        
        record, bit = n, 1
        
        while record:
            n ^= bit
            bit <<= 1
            record >>= 1
            
        return n